packages/firestore/test/unit/specs/listen_spec.test.ts

```diff
@@ -509,4 +509,30 @@ describeSpec('Listens:', [], () => {
       .watchAcksFull(query, 3000)
       .expectEvents(query, {});
   });
+
+  specTest('Persists global resume tokens', [], () => {
+    const query = Query.atPath(path('collection'));
+    const docA = doc('collection/a', 1000, { key: 'a' });
+
+    return (
+      spec()
+        .withGCEnabled(false)
+        .userListens(query)
+        .watchAcksFull(query, 1000, docA)
+        .expectEvents(query, { added: [docA] })
+
+        // Some time later, watch sends an updated resume token and the user stops
+        // listening.
+        .watchSnapshots(2000, [], 'resume-token-2000')
+        .userUnlistens(query)
+        .watchRemoves(query)
+
+        .userListens(query, 'resume-token-2000')
+        .expectEvents(query, { added: [docA], fromCache: true })
+        .watchAcks(query)
+        .watchCurrents(query, 'resume-token-3000')
+        .watchSnapshots(3000)
+        .expectEvents(query, { fromCache: false })
+    );
+  });
```

Comment:
It seems pretty straightforward to test the MAX_RESUME_TOKEN_BUFFERING_MICROS logic too if we wanted to (might need to use .restart() to test re-listening without unlistening).

Reply:
Okay fine, you caught my lazy bones out. Done.